base/win/comptr.h

```diff
@@ -31,6 +33,14 @@ namespace base {
       std::swap(m_ptr, tmp.m_ptr);
     }
 
+    ComPtr(T* p) {
+      if (!p)
+        return;
+
+      if (p->QueryInterface<T>(&m_ptr) != NOERROR)
+        m_ptr = nullptr;
+    }
+
```

Comment:
As `T` is the same as the `T*p`, I think we can simplify this as
```suggestion
    explicit ComPtr(T* p) : m_ptr(p) {
      if (m_ptr)
        m_ptr->AddRef();
    }

```
I have added the `explicit` keyword to avoid implicit conversion between raw pointers and this class adding new references/querying interfaces.

Reply:
This simplification actually skips the [QueryInterface](https://learn.microsoft.com/en-us/windows/win32/api/unknwn/nf-unknwn-iunknown-queryinterface(q)) call...which I actually wanted to call because of the validations it does. Do you they are not needed?

Reply:
If you are copying/adding a new reference to a COM object pointer of the same type (T*) the `QueryInterface()` is not needed (shouldn't), as the pointer to `T` interface should be already valid. (`QueryInterface` is used to ask for another interface.)

Unrelated to this: The only problem with this kind of constructor (`ComPtr(T*)`) is that we might want two kind of behaviors: 1) adding a new reference (with AddRef), or 2) wrapping an existent reference to do just the automatic release (e.g. as clip comptr).

Anyway the `AddRef()` behavior is the most common, and in case you need one without AddRef we could add something like an attach method (e.g. https://github.com/microsoft/wil/blob/5c6a7ba43e79edd71dcd1ce4c2d21b0610183c86/include/wil/com.h#L360 ).